libcalico-go/lib/client/config.go

```diff
@@ -152,15 +152,15 @@ func (c *config) GetGlobalASNumber() (numorstring.ASNumber, error) {
 // IP in IP tunnel.
 func (c *config) SetGlobalIPIP(enabled bool) error {
 	_, err := c.c.Backend.Apply(context.Background(), &model.KVPair{
-		Key:   model.GlobalConfigKey{Name: "IpInIpEnabled"},
+		Key:   model.GlobalConfigKey{Name: "DeprecatedIpInIpEnabled"},
```

Comment:
I'm very against all this field renaming, I think you should back it all out.  If you disagree then lets have a chat and settle the matter.  If you agree then please grep your code for the "DeprecatedXXX" naming and remove it all.

Some reasons why I'm against it:
* In the public APIs it means a user has to make a change to keep the same behaviour at upgrade.  That adds unacceptable friction (I;d prefer to remove the old fields entirely).  A change to the felixconfig CRD would mean that editing the resource with old and new calicoctl would do different things and we'd need to add conversion logic to handle upgrade.
* In the model API, it creates a breaking change in the felix/typha API so, over upgrade you'll get an unnecessary mismatch.

Reply:
I agree, this is something I missed rolling back. I'll be more thorough going through the diff to make sure I don't miss anything. Sorry and thanks.